src/main.rs

```diff
@@ -136,33 +137,110 @@ fn gather_metrics() -> Vec<u8> {
     buffer
 }
 
-// TODO: Implement
-fn _storage_backend<S>(m: &clap::ArgMatches) -> Box<dyn (Fn() -> S) + Send> {
+// Creates the filesystem storage back-end
+fn fs_storage_backend(m: &clap::ArgMatches) -> Box<dyn (Fn() -> libunftp::storage::filesystem::Filesystem) + Send> {
+    let p: PathBuf = m.value_of(args::ROOT_DIR).unwrap().into();
+    Box::new(move || libunftp::storage::filesystem::Filesystem::new(p.clone()))
```

Comment:
Its because its an Fn closure.  Without the clone you get:

```
cannot move out of `p`, a captured variable in an `Fn` closure
```

with the clone but not the move you get:

```
 closure may outlive the current function, but it borrows `p`, which is owned by the current function
```